apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.Data/SpannerTransaction.cs

```diff
@@ -149,12 +150,14 @@ private void CheckCompatibleMode(TransactionMode mode)
             switch (mode)
             {
                 case TransactionMode.ReadOnly:
-                    Mode.AssertTrue(x => x == TransactionMode.ReadOnly || x == TransactionMode.ReadWrite,
-                        "You can only execute reads on a ReadWrite or ReadOnly Transaction!");
+                {
+                    GaxPreconditions.CheckState(((Predicate<TransactionMode>) (x => x == TransactionMode.ReadOnly || x == TransactionMode.ReadWrite))(Mode), "You can only execute reads on a ReadWrite or ReadOnly Transaction!");
```

Comment:
I went thru and simplified them all.  I did an inline refactor and that made them unnecessarily complex.